src/settings/mod.rs

```diff
@@ -38,22 +39,53 @@ pub mod types;
 
 const CARGO_PKG_VERSION: &str = env!("CARGO_PKG_VERSION");
 
+#[derive(Debug)]
+pub struct AllSettings {
+    pub cli: CliSettings,
+    pub lib: Settings,
+}
+
+impl AllSettings {
+    pub fn from_configuration(config: Configuration, project_root: &Path) -> Result<Self> {
+        Ok(Self {
+            cli: CliSettings {
+                cache_dir: config
```

Comment:
Should we impl `From<&Configuration>`?

Reply:
We'd have to get rid of the additional `project_root` parameter ... I am not sure if that's desirable.